### PR TITLE
Refactor response from Kuzzle

### PR DESCRIPTION
**Pull request by AnthonySendra**
> **CAUTION**: This PR introduces a lot of breaking changes. Don't merge this PR untill the SDK is ready. And when the merge is done, don't forget to change the minor version.
> Also check if the API document is up to date with @benoitvidis branch [kuz352-responseobject-refactor](https://github.com/kuzzleio/kuzzle-api-documentation/tree/kuz352-responseobject-refactor)

Refactor the response from Kuzzle in order to have a consistent response like the following object:

* When we have just one element (e.g with a get):
```json
{
    "action": "get",
    "collection": "user",
    "controller": "read",
    "error": null,
    "index": "mainindex",
    "metadata": {},
    "requestId": "a39f348b-f73d-4518-9cb4-32a4b0dc2040",
    "result": {
        "_id": "AVItAloUTHHcXT3l6WXg",
        "_index": "mainindex",
        "_source": {
            "username": "toto"
        },
        "_type": "user",
        "_version": 1,
        "found": true
    },
    "scope": null,
    "state": "done",
    "status": 200
}
```

* When we have a list (e.g getAllStats)
```json
{
    "action": "getAllStats",
    "controller": "admin",
    "error": null,
    "metadata": {},
    "requestId": "608fef72-da25-48e1-8b77-ae351d33bae2",
    "result": {
        "hits": [
            {
                "completedRequests": {
                    "rest": 1
                },
                "connections": {},
                "failedRequests": {},
                "ongoingRequests": {
                    "rest": 0
                },
                "timestamp": "2016-01-10T19:30:06.328Z"
            }
        ],
        "total": 1
    },
    "scope": null,
    "state": "done",
    "status": 200
}
```

We have now a result directly in `result` entry with the object requested or with an entry `hits` which is an array.

***Note:*** 
* We have no more `hits.hits` in `result` object. All previous entries in `hits` are now in root object `result`, i.e:
```json
"result": {
        "_shards": {
            "failed": 0,
            "successful": 5,
            "total": 5
        },
        "hits": [
            {
                "_id": "AVItE6_7THHcXT3l6WZH",
                "_index": "mainindex",
                "_score": 1,
                "_source": {
                    "username": "toto"
                },
                "_type": "user"
            }
        ],
        "max_score": 1,
        "timed_out": false,
        "took": 1,
        "total": 1
    }
```
* The `error` entry can contains a `_source` to find out which query failed.

### CODE REVIEW

File: lib/api/core/models/responseObject.js [hunk 1]
```diff
@@ -57,22 +57,29 @@ function construct (requestObject, response) {
   this.metadata = requestObject.metadata;
   this.state = requestObject.state || 'done';
 
-
-  this.data = !util.isError(response) ? _.extend(requestObject.data, response) : requestObject.data;
+  // The real document content is past in 'body' attribute in order to be iso with RequestObject for dsl.testFilters function,
```
Comment (scottinet) [inline lib/api/core/models/responseObject.js]: nitpicking: "the real document content is passed in the 'body' attribute"

**Comment by scottinet** [comment]
Very nice!
I'll prepare a new Javascript SDK version right away

+1

**Comment by ballinette** [comment]
+1

**Comment by jenow** [comment]
+1

**Comment by benoitvidis** [comment]
+1

**Pull request merged**